test/unit/view.js

```diff
@@ -5,67 +5,105 @@ import Iframe from '../../src/iframe';
       assert.instanceOf(view.template, Template);
     });
+
+    it('sets eventBound to false', () => {
+      assert.equal(view.eventsBound, false);
+    });
   });
 
   describe('prototype methods', () => {
+    let component;
+    let view;
+
+    beforeEach(() => {
+      Component.prototype.typeKey = 'product';
+      component = new Component({
+        id: 1234,
+        node: document.createElement('div'),
+      }, {browserFeatures: {}});
+      view = new View(component);
+    });
+
     describe('init()', () => {
-      let component;
-      let view;
+      let loadStub;
+      let addClassStub;
+      const loadRes = 'done';
 
       beforeEach(() => {
-        component = new Component({
-          id: 1234,
-          node: document.createElement('div'),
-        }, {browserFeatures: {}});
-        view = new View(component);
-        component.typeKey = 'product';
+        loadStub = sinon.stub(Iframe.prototype, 'load').resolves(loadRes);
+        addClassStub = sinon.stub(Iframe.prototype, 'addClass');
+        component = Object.defineProperty(component, 'options', {
+          writable: true,
+          value: {
+            iframe: true,
+            manifest: ['product', 'option'],
+          },
+        });
       });
 
-      it('returns a promise if iframe option is false', () => {
-        component.config.product.iframe = false;
-        return view.init().then((iframe) => {
-          assert.isNull(iframe);
-        });
+      afterEach(() => {
+        loadStub.restore();
+        addClassStub.restore();
       });
 
-      it('loads an Iframe', () => {
-        component.config.product.iframe = true;
-        const loadStub = sinon.stub(Iframe.prototype, 'load').returns(Promise.resolve());
-        const addClassSpy = sinon.spy(Iframe.prototype, 'addClass');
-        return view.init().then(() => {
-          assert.instanceOf(view.iframe, Iframe);
-          assert.equal(component.node.className, ' shopify-buy-frame shopify-buy-frame--product');
-          assert.calledOnce(loadStub);
-          assert.calledWith(addClassSpy, '');
-          loadStub.restore();
-          addClassSpy.restore();
-        });
+      it('returns a promise if iframe option is false', async () => {
+        component.options.iframe = false;
+        const iframe = await view.init();
+        assert.isNull(iframe);
       });
-    });
 
-    describe('render()', () => {
-      let component;
-      let view;
+      it('returns the iframe if it already exists', async () => {
+        view.iframe = 'iframe';
+        const iframe = await view.init();
+        assert.equal(iframe, view.iframe);
+      });
 
-      beforeEach(() => {
-        component = new Component({
-          id: 1234,
-          node: document.createElement('div'),
-        }, {browserFeatures: {}});
-        view = new View(component);
-        component.typeKey = 'product';
+      it('create, and loads an returns an Iframe', async () => {
```

Comment:
Were not testing the return in this test, so maybe say something like `creates and loads an Iframe`.